client/packages/common/src/ui/forms/JsonForms/components/EncounterLineChart.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import { rankWith, uiTypeIs, ControlProps } from '@jsonforms/core';
+import { withJsonFormsControlProps } from '@jsonforms/react';
+import { Box } from '@mui/material';
+import {
+  Line,
+  LineChart,
+  useDocument,
+  useFormatDateTime,
+  XAxis,
+  YAxis,
+} from '@openmsupply-client/common';
+
+import { CartesianGrid, Tooltip, TooltipProps } from 'recharts';
+import { Label } from 'recharts';
+
+export const encounterLineChartTester = rankWith(
+  4,
+  uiTypeIs('EncounterLineChart')
+);
+
+type Options = {
+  values?: ValueOption[];
+};
+
+type ValueOption = {
+  field?: string;
+  label?: string;
+  unit?: string;
+};
+
+type DateTimeTooltipProps = TooltipProps<string, string> & {
+  name: string;
+  unit: string;
+};
+
+const DateTimeTooltip = (props: DateTimeTooltipProps) => {
+  const { localisedDateTime } = useFormatDateTime();
+
+  if (!props.active || props.payload?.[0] === undefined) {
+    return null;
+  }
+  return (
+    <Box padding={0.5} border={1} borderColor={'lightgray'}>
+      <p>{`${localisedDateTime(props.label)}`}</p>
+      <p>{`${props.name}: ${props.payload[0].value} [${props.unit}]`}</p>
+    </Box>
+  );
+};
+
+type DataType = { time: number; y: number };
+
+const UIComponent = (props: ControlProps) => {
+  const { dayMonthShort } = useFormatDateTime();
+
+  const option = (props.uischema.options as Options | undefined)?.values?.[0];
+  const [data, setData] = useState([] as DataType[]);
+  const { data: extractedData } = useDocument.encounterExtractFields(
```

Comment:
is there some way to pass in this data instead?
the `common` package shouldn't be calling domain specific methods like this - this usage is very specific to the app, whereas the intention of 'common' is that it can be used with other apps.

Reply:
I moved all program related controls out of common:
- useJsonForms (because of registry api call)
- DateOfBirth
- EncounterLineChart
- IdGenerator